plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/IcebergRESTCatalogConfig.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.rest;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+
+import java.util.Optional;
+
+public class IcebergRESTCatalogConfig
+{
+    private String credential;
+    private String token;
```

Comment:
We generally introduce an enum class when supporting several auth types likes `ElasticsearchConfig`

Reply:
These are both the same auth type (OAuth2) and the credential vs. token just determines what flow will be used.  Currently there isn't support for other auth types (Basic, Kerberos, etc.), but if we do add those, I agree it would make sense to add an enum.

Reply:
Does REST catalog support no-auth? If yes, it would be better to introduce enum class.

Reply:
added an enum with `NONE` / `OAUTH2` options